service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,47 @@
         </actions>
     </service>
 
+    <service verb="get" noun="PackageInfo">
+        <implements service="mantle.shipment.CarrierServices.get#AutoPackageInfo"/>
+        <actions>
+            <!--Get shipmentIds from ShipmentAndItemSource using orderId and orderItemSeqId of itemInfoList-->
+            <set field="shipmentIdsList" from="new HashSet()"/>
+            <iterate list="itemInfoList" entry="itemInfoMap">
+                <entity-find-one entity-name="mantle.shipment.ShipmentAndItemSource" value-field="shipmentSourceList">
+                    <field-map field-name="orderId" from="itemInfoMap.orderId"/>
+                    <field-map field-name="orderItemSeqId" from="itemInfoMap.orderItemSeqId"/>
+                </entity-find-one>
+                <script>shipmentIdsList.add(shipmentSourceList.shipmentId)</script>
```

Comment:
There is a way to collect all shipmentIds in set using the list.
Please check below for reference:
https://github.com/hotwax/DevScreens/blob/master/screen/SimpleScreens/Catalog/Product/EditProduct.xml#L89